lib/providers/notifiers/connection_notifier.dart

```diff
@@ -13,31 +19,48 @@ class ConnectionNotifier extends ChangeNotifier {
     'https://mainnet.telosusa.io',
     'https://telos.eosphere.io',
     'https://telos.caleos.io',
+    'https://api.eos.miami',
+    'https://hyperion.telosgermany.io',
   ];
 
   void init() {
     discoverEndpoints();
   }
 
   void discoverEndpoints() async {
+    List<Future> checks = List<Future>();
+
     for (var endpoint in availableEndpoints) {
-      var ping = Stopwatch()..start();
+      checks.add(checkEndpoint(endpoint));
+    }
 
-      Response res = await get("$endpoint/v2/health");
+    var responses = await Future.wait(checks);
 
-      ping.stop();
+    responses.sort((a, b) => a.ping - b.ping);
 
-      if (res.statusCode == 200) {
-        int endpointPing = ping.elapsedMilliseconds;
+    //for (int i=0; i<responses.length; i++) {
+    //  print("$i ping: ${responses[i].ping} endpoint: "+responses[i].url);
+    //}
+    currentEndpoint = responses[0].url;
+    currentEndpointPing = responses[0].ping;
+    notifyListeners();
 
-        print("ping from $endpoint is $endpointPing");
+  }
 
-        if (currentEndpointPing == 0 || endpointPing < currentEndpointPing) {
-          currentEndpoint = endpoint;
-          currentEndpointPing = endpointPing;
-          notifyListeners();
-        }
+  Future<Endpoint> checkEndpoint(String endpoint) async {
+    try {
+      var ping = Stopwatch()..start();
+      Response res = await get("$endpoint/v2/health");
+      ping.stop();
+      if (res.statusCode == 200) {
+        int endpointPing = ping.elapsedMilliseconds;
+        return Endpoint(endpoint, endpointPing);
+      } else {
+        return Endpoint(endpoint, 1000000);
```

Comment:
Magic numbers. 

why 1000000? 
If this number is placed in a variable it can be given a meaningful name. Helping others why we are returning this number.

Reply:
It's an endpoint with an infinite timeout

Reply:
named now - yes we could store an error flag but it would just add an additional storage byte, and make comparing code more complicated and error prone. So in this case I do like the infinite ping.

Reply:
>additional storage byte
Storage is cheap. Very cheap, we are not writing assembly code. 

>comparing code more complicated and error prone
The opposite, i think it increases readability and maintainability.

`return Endpoint(endpoint, endpointPing);`
`return Endpoint(endpoint, 1000000);`
`return Endpoint(endpoint, 2000000);`

^^ At first glance reading this code I cant figure out why 200000 over 1000000? 
I still dont know why one over the other. 


`return Endpoint(endpoint, endpointPing);`
`return Endpoint(endpoint, INFINITE_PING_WHEN_ERROR_CODE);`
`return Endpoint(endpoint, INFINITE_PING_WHEN_CRASH);`

Reading this code, I have a better understanding of these numbers and the checkEndpoint function overall. 

Furthermore: Down the line you need to check if the ping is xxx or 10000000 or 2000000
Instead of comparing numbers, you can compare using the constants. 

val endpoint = checkEndpoint(www.miami.com) 

If(endpoint.ping == INFINITE_PING_WHEN_ERROR_CODE ) {
   foo 
} else {
   bar
}

Multiple languages lint rules use no-magic-number.
 
**I am not blocking the PR for this tho.**

https://palantir.github.io/tslint/rules/no-magic-numbers/